app/celery/service_callback_tasks.py

```diff
@@ -17,44 +17,78 @@
 
 @notify_celery.task(bind=True, name="send-delivery-status", max_retries=5, default_retry_delay=300)
 @statsd(namespace="tasks")
-def send_delivery_status_to_service(self, notification_id,
-                                    encrypted_status_update
-                                    ):
-    try:
-        status_update = encryption.decrypt(encrypted_status_update)
+def send_delivery_status_to_service(
+    self, notification_id, encrypted_status_update
+):
+    status_update = encryption.decrypt(encrypted_status_update)
+
+    data = {
+        "notification_id": str(notification_id),
+        "reference": status_update['notification_client_reference'],
+        "to": status_update['notification_to'],
+        "status": status_update['notification_status'],
+        "created_at": status_update['notification_created_at'],
+        "completed_at": status_update['notification_updated_at'],
+        "sent_at": status_update['notification_sent_at'],
+        "notification_type": status_update['notification_type']
+    }
+    _send_data_to_service_callback_api(
+        self,
+        data,
+        status_update['service_callback_api_url'],
+        status_update['service_callback_api_bearer_token'],
+        'send_delivery_status_to_service'
+    )
+
+
+@notify_celery.task(bind=True, name="send-complaint", max_retries=5, default_retry_delay=300)
+@statsd(namespace="tasks")
+def send_complaint_to_service(self, complaint_data):
+    complaint = encryption.decrypt(complaint_data)
+
+    data = {
+        "notification_id": complaint['notification_id'],
+        "complaint_id": complaint['complaint_id'],
+        "reference": complaint['reference'],
+        "to": complaint['to'],
+        "complaint_date": complaint['complaint_date']
+    }
+
+    _send_data_to_service_callback_api(
+        self,
+        data,
+        complaint['service_callback_api_url'],
+        complaint['service_callback_api_bearer_token'],
+        'send_complaint_to_service'
+    )
 
-        data = {
-            "id": str(notification_id),
-            "reference": status_update['notification_client_reference'],
-            "to": status_update['notification_to'],
-            "status": status_update['notification_status'],
-            "created_at": status_update['notification_created_at'],
-            "completed_at": status_update['notification_updated_at'],
-            "sent_at": status_update['notification_sent_at'],
-            "notification_type": status_update['notification_type']
-        }
 
+def _send_data_to_service_callback_api(self, data, service_callback_url, token, function_name):
+    notification_id = data["notification_id"]
```

Comment:
`complaint_id` would be probably more useful than `notification_id` for complaint callback logs. Having to pass through all the variables for log message is a bit unwieldy, but we could do something like `object_id = data.get('id') or data.get('complaint_id')` here 🤷‍♂️

Alternatively, we could set `'id': str(complaint.id)` in complaint data and then the main key is the same in both payloads.

Reply:
We would need to send the notification_id to the client. It's more relevant than the complaint id is to them, as some clients do not set the client_reference

Reply:
We do send both to the service, I think this variable is only used for the logs and it feels like it'd be easier for us to lookup complaint data by `complaint_id` rather than `notification_id`

Reply:
ok, decision made, we are leaving it as is 👍